internal/pkg/poster/poster.go

```diff
@@ -70,148 +81,146 @@ func (p *Poster) StopPoSting() {
+		return
+	}
+
+	// Some day we might want to choose a subset of partitions to prove at one time. Today is not that day.
+	sectors, err := stateView.MinerSectorInfoForDeadline(ctx, p.minerAddr, deadlineIndex, partitions)
+	if err != nil {
+		log.Errorf("error retrieving sector info for miner %s partitions at index %d: %s", p.minerAddr, deadlineIndex, err)
+		return
+	}
+
+	// if no sectors, we're done
+	if len(sectors) == 0 {
+		return
+	}
+
+	buf := new(bytes.Buffer)
+	err = p.minerAddr.MarshalCBOR(buf)
+	if err != nil {
+		log.Errorf("could not create entropy from miner address %s: %s", p.minerAddr, err)
+		return
+	}
+
+	randomness, err := p.chain.SampleChainRandomness(ctx, head, acrypto.DomainSeparationTag_WindowedPoStChallengeSeed, challengeAt, buf.Bytes())
+	if err != nil {
+		log.Errorf("could not sample chain randomness at %d: %s", challengeAt, err)
+		return
+	}
+
+	proofs, err := p.mgr.GenerateWindowPoSt(ctx, abi.ActorID(minerID), sectors, abi.PoStRandomness(randomness))
+	if err != nil {
+		log.Errorf("error generating window PoSt: %s", err)
+		return
+	}
+
+	_, workerAddr, err := stateView.MinerControlAddresses(ctx, p.minerAddr)
+	if err != nil {
+		log.Errorf("could not get miner worker address fro miner %s: %s", p.minerAddr, err)
+		return
+	}
+
+	err = p.sendPoSt(ctx, workerAddr, deadlineIndex, partitions, proofs)
+	if err != nil {
+		log.Error("error sending window PoSt: ", err)
+		return
+	}
 }
 
-func (p *Poster) getChallengeSeed(ctx context.Context, head block.TipSetKey, height abi.ChainEpoch, minerAddr address.Address) ([32]byte, error) {
-	var challengeSeed [32]byte
+func (p *Poster) sendPoSt(ctx context.Context, workerAddr address.Address, index uint64, partitions []uint64, proofs []abi.PoStProof) error {
+
+	windowedPost := &miner.SubmitWindowedPoStParams{
+		Deadline:   index,
+		Partitions: partitions,
+		Proofs:     proofs,
+		Skipped:    abi.BitField{},
+	}
 
-	entropy, err := encoding.Encode(minerAddr)
+	mcid, errCh, err := p.outbox.Send(
+		ctx,
+		workerAddr,
+		p.minerAddr,
+		types.ZeroAttoFIL,
+		types.NewGasPrice(1),
+		gas.NewGas(10000),
+		true,
+		builtin.MethodsMiner.SubmitWindowedPoSt,
+		windowedPost,
+	)
 	if err != nil {
-		return challengeSeed, err
+		return err
 	}
-	randomness, err := p.chain.SampleChainRandomness(ctx, head, acrypto.DomainSeparationTag_WindowedPoStChallengeSeed, height, entropy)
+	if err := <-errCh; err != nil {
+		return err
+	}
+
+	// wait until we see the post on chain at least once
+	err = p.waiter.Wait(ctx, mcid, func(_ *block.Block, _ *types.SignedMessage, recp *vm.MessageReceipt) error {
+		return nil
+	})
 	if err != nil {
-		return challengeSeed, err
+		return err
 	}
 
-	copy(challengeSeed[:], randomness)
-	return challengeSeed, nil
+	return nil
+}
+
+func (p *Poster) getProvingSet(ctx context.Context, stateView *appstate.View) ([]abi.SectorInfo, error) {
+	return consensus.NewPowerTableView(stateView, stateView).SortedSectorInfos(ctx, p.minerAddr)
```

Comment:
Thankfully, this method seems to be unused. We should only load the sector info for sectors being proven.

Reply:
removed